Performance enhancements

zhjngli opened the pull request.

zhjngli committed:
Remove fadein, add no js plugin

ghost commented:
# [Gatsby Cloud](https://www.gatsbyjs.com) Build Report
**personal-website-main**

:tada: Your build was successful! [See the Deploy preview here.](https://personalwebsitemain64908-fadeinnojs.gtsb.io)

## Build Details

[View the build logs here.](https://gatsbyjs.com/dashboard/cefb9456-dc41-4617-9700-a207f1fe0cc1/sites/bec16e66-2c27-405f-8217-fa058d311730/builds/85e0a123-7354-44dc-8a27-91793992acd5/details)

:clock1: Build time: **3m**

## Performance

### Lighthouse report

| Metric | Score |
| ----- | ---- |
| Performance |   :green_heart: 100   |
| Accessibility |   :green_heart: 96   |
| Best Practices |   :green_heart: 100   |
| SEO |   :green_heart: 91   |

:link: [View full report](https://build-85e0a123-7354-44dc-8a27-91793992acd5.gtsb.io/reports/lighthouse/index.html)

zhjngli commented:
![nojs](https://user-images.githubusercontent.com/3613177/131641465-322088eb-b991-4c80-8e0e-0da202ff9b99.png)


![yesjs](https://user-images.githubusercontent.com/3613177/131641475-014ff150-54a8-4367-9e68-0914cf8e1f78.png)

it does load significantly faster, however, the experience feels clunky as there's no fade in. consider excluding the fadein from the no js plugin.

zhjngli committed:
Fix formatting in layout

zhjngli committed:
Remove react-fade-in completely

zhjngli commented:
consider removing less stuff, it transfers less resources, but without the js, it feels less like a SPA and more like lots of html files chained together

zhjngli committed:
Exclude js from no js plugin

Attempts to avoid loading page-data and other useless things

zhjngli committed:
Use new no js plugin

zhjngli committed:
Keep scripts for proper routing

zhjngli committed:
Remove gatsby-plugin-preload-fonts

Not helpful for FOUT anyway. Only decreases performance by blocking on
loading fonts

zhjngli commented:
Performance summary:
- no js helps a LOT but it also removes the nice feel of a SPA
- removing fade in is nice for instant loads. not as sweet of a UX but i could reduce the fade in ms later
- preload fonts doesn't help at all. it blocks loading and doesn't fix FOUT either
